packages/amplify_analytics_pinpoint/lib/amplify_analytics_pinpoint.dart

```diff
@@ -37,7 +37,7 @@ class AmplifyAnalyticsPinpoint extends AnalyticsPluginInterface {
   /// class that extends [AnalyticsPluginInterface] when they register themselves.
   static set instance(AnalyticsPluginInterface instance) {
     PlatformInterface.verifyToken(instance, _token);
-    _instance = instance;
+    _instance = instance as AmplifyAnalyticsPinpoint;
```

Comment:
I left a similar comment on #485. Any reason not to update the setter/getter to be of type `AmplifyAnalyticsPinpoint` instead of casting?

```dart
  static AmplifyAnalyticsPinpoint get instance => _instance;

  static set instance(AmplifyAnalyticsPinpoint instance) {
    PlatformInterface.verifyToken(instance, _token);
    _instance = instance;
  }
```

Reply:
Good point, we shouldn't be casting the instance parameter of type AnalyticsPluginInterface to AmplifyAnalyticsPinpoint if we're going to be saving it as type AnalyticsPluginInterface anyways!

Reply:
`  static set instance(AnalyticsPluginInterface instance) {
    PlatformInterface.verifyToken(instance, _token);
    _instance = instance;
  }`

We should be saving as Interface type though to allow for others to provide their own interface implementation though.